### ISSUE TITLE
Need to add `CGColorSpaceRelease` calls?

### ISSUE THREAD

**Issue by tmm1** [KhronosGroup/MoltenVK#940]
I was looking at the swapchain colorspace code:

https://github.com/KhronosGroup/MoltenVK/blob/21630b59bb65ae6a329186786b5eb02c2a37f881/MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm#L284-L294

and noticed how it differs from the example on https://developer.apple.com/documentation/metal/drawable_objects/displaying_hdr_content_in_a_metal_layer/using_color_spaces_to_display_hdr_content?language=objc

```
const CFStringRef name = kCGColorSpaceITUR_2020_PQ_EOTF;
CGColorSpaceRef colorspace = CGColorSpaceCreateWithName(name);
metalLayer.colorspace = colorspace;
CGColorSpaceRelease(colorspace);
```

Are we leaking references without the `CGColorSpaceRelease` calls?

**Comment by cdavis5e** [comment]
I wasn't sure if the `colorspace` property retained the object assigned to it--it isn't marked `retain` in the header--so I assumed it was better to let it leak than to over-release the object. The sample code you linked to suggests that it does in fact retain the object.

**Comment by KenThomases** [comment]
Memory management is designed to be a local concern. You're supposed to do the right thing in your code and assume that other code is doing the right thing. If `CAMetalLayer` is keeping a reference to an object, it's its responsibility to retain it. Just assume it handles its responsibilities.

A non-Objective-C-object property can't use `retain`, so that's not informative.

**Comment by billhollings** [comment]
@cdavis Fairly simple fix. I'll take care of it if you haven't started on it yet.

**Comment by tmm1** [comment]
Perhaps a private extension that adds `setColorspaceName:`?

**Comment by billhollings** [comment]
>Perhaps a private extension that adds `setColorspaceName:`?

Not a bad idea. And MoltenVK already has a private extension for `CAMetalLayer` in `CAMetalLayer+MoltenVK.h`.

**Comment by cdavis5e** [comment]
> @cdavis Fairly simple fix. I'll take care of it if you haven't started on it yet.

I have not. Also, wrong cdavis. 😄

**Comment by billhollings** [comment]
Fixed in PR #947.
